src/best_of/integrations/gitlab_integration.py

```diff
@@ -0,0 +1,212 @@
+import logging
+
+import requests
+from addict import Dict
+from dateutil.parser import parse
+
+from best_of import utils
+from best_of.default_config import MIN_PROJECT_DESC_LENGTH
+from best_of.integrations.base_integration import BaseIntegration
+
+log = logging.getLogger(__name__)
+
+query = """
+query($fullPath: ID!) {
+  project(fullPath: $fullPath) {
+    name
+    forksCount
+    starCount
+    issueStatusCounts {
+      all
+      closed
+      opened
+    }
+    description
+    createdAt
+    lastActivityAt
+    mergeRequests {
+      count
+    }
+    webUrl
+    httpUrlToRepo
+    statistics {
+      commitCount
+    }
+    releases(first: 100, sort: CREATED_DESC) {
+      edges {
+        node {
+          createdAt
+          name
+          releasedAt
+          tagName
+          tagPath
+          upcomingRelease
+        }
+      }
+    }
+  }
+}
+"""
+
+GITLAB_DEFAULT_API = "https://gitlab.com/api/graphql"
+
+
+class GitLabIntegration(BaseIntegration):
+    @property
+    def name(self) -> str:
+        return "gitlab"
+
+    def update_project_info(self, project_info: Dict) -> None:
+
+        # project_info:
+        # gitlab_url, homepage, name, gitlab_id, license, created_at
+        # updated_at, last_commit_pushed_at, fork_count, open_issue_count
+        # closed_issue_count, star_count, commit_count,
+        # latest_stable_release_published_at, latest_stable_release_number
+        # github_release_downloads, monthly_downloads, release_count
+        # description, dependent_project_count, contributor_count
+
+        api_url = project_info.api_url or GITLAB_DEFAULT_API
+        variables = {"fullPath": project_info.gitlab_id}
+        try:
+            request = requests.post(
+                api_url, json={"query": query, "variables": variables}
+            )
+
+            if request.status_code != 200:
+                log.info(
+                    f"Unable to find the repo {project_info.gitlab_id}. Statuscode: {request.status_code}"
+                )
+                return
+
+            repo_info = Dict(request.json()["data"]["project"])
+        except Exception as ex:
+            log.info(
+                f"Failed to request the repo {project_info.gitlab_id} on API {api_url} ",
+                exc_info=ex,
+            )
+            return
+
+        if not project_info.gitlab_url and repo_info.httpUrlToRepo:
+            project_info.gitlab_url = repo_info.httpUrlToRepo
+
+        if not project_info.homepage:
+            project_info.homepage = repo_info.webUrl
```

Comment:
What is the `webURL`, is it the URL to the repo? What is the difference to `httpUrlToRepo`?

Reply:
Do you know if `webURL` is always present? Otherwise, we might add a fallback to `httpUrlToRepo` here as well since the `homepage` attribute is required.

Reply:
It looks like `webURL` is set to `httpUrlToRepo` by default in the API